### DISCUSSION TITLE
关于bulkexecutor的疑问

### DISCUSSION THREAD

**Discussion by zhouhanAC** [zeromicro/go-zero#1270] [Q&A] [answered]
任务累计>=阈值才会执行，那任务数不等于阈值倍数，比如阈值为10，任务数为49，只执行了前40个任务，是不是不太合理

**Comment by kevwan** [comment, marked as answer]
有定期执行的，默认1s，最后9个也会执行。

**Reply by zhouhanAC** [reply]
原来是在Flush()中强制执行没有完成的任务，是TestBulkExecutor这个单测没有调用Flush，看漏了，感谢大佬指点

**Reply by kevwan** [reply]
不客气，因为单测没法定时任务哈
